artifactory/commands/transferfiles/delayedartifactshandler.go

```diff
@@ -0,0 +1,209 @@
+package transferfiles
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/jfrog/jfrog-client-go/utils/errorutils"
+	"github.com/jfrog/jfrog-client-go/utils/io/content"
+	"github.com/jfrog/jfrog-client-go/utils/log"
+	"os"
+	"path"
+	"path/filepath"
+)
+
+const (
+	// Max delayedArtifacts that will be written in a file
+	maxDelayedArtifactsInFile = 50000
+)
+
+// TransferDelayedArtifactsMng managing multi threads writing delayed deployment artifacts in order to keep the order of deployment.
```

Comment:
// TransferDelayedArtifactsMng managing multi threads writing delayed deployment artifacts in order to keep the order of deployment.
-->
// TransferDelayedArtifactsMng takes care of the multi threaded writing of artifacts to be transferred. while maintaining the correct order of the deployment. This is needed because, for example, for maven repositories, pom file should be deployed last.